src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.Shared.cs

```diff
@@ -0,0 +1,281 @@
+        //Leaving these constants same as they are defined in Windows
+        internal const int ALG_TYPE_DSS = (1 << 9);
+        internal const int ALG_TYPE_RSA = (2 << 9);
+        internal const int ALG_TYPE_BLOCK = (3 << 9);
+        internal const int ALG_CLASS_SIGNATURE = (1 << 13);
+        internal const int ALG_CLASS_DATA_ENCRYPT = (3 << 13);
+        internal const int ALG_CLASS_KEY_EXCHANGE = (5 << 13);
+        internal const int CALG_RSA_SIGN = (ALG_CLASS_SIGNATURE | ALG_TYPE_RSA | 0);
+        internal const int CALG_DSS_SIGN = (ALG_CLASS_SIGNATURE | ALG_TYPE_DSS | 0);
+        internal const int CALG_RSA_KEYX = (ALG_CLASS_KEY_EXCHANGE | ALG_TYPE_RSA | 0);
+
+        internal const int ALG_CLASS_HASH = (4 << 13);
+        internal const int ALG_TYPE_ANY = (0);
+
+        internal const int SIMPLEBLOB = 0x1;
+        internal const int PUBLICKEYBLOB = 0x6;
+        internal const int PRIVATEKEYBLOB = 0x7;
+        internal const int PLAINTEXTKEYBLOB = 0x8;
+        internal const byte BLOBHEADER_CURRENT_BVERSION = 0x2;
+        internal const int CRYPT_BLOB_VER3 = 0x00000080; // export version 3 of a blob type
+        internal const int RSA_PUB_MAGIC = 0x31415352;
+        internal const int RSA_PRIV_MAGIC = 0x32415352;
+
+        // Provider type to use by default for RSA operations. We want to use RSA-AES CSP
+        // since it enables access to SHA-2 operations. All currently supported OSes support RSA-AES.
+        internal const int DefaultRsaProviderType = (int)ProviderType.PROV_RSA_AES;
+
+        internal enum CryptKeyError : uint
+        {
+            NTE_NO_KEY = 0x8009000D, // Key does not exist.
+            NTE_BAD_KEYSET = 0x80090016, // Keyset does not exist.
+            NTE_KEYSET_NOT_DEF = 0x80090019, // The keyset is not defined.
+            NTE_BAD_KEY_STATE = 0x8009000B,
+            NTE_BAD_KEYSET_PARAM = 0x8009001F,
+            NTE_KEYSET_ENTRY_BAD = 0x8009001A,
+            NTE_FILENOTFOUND = 0x80070002,
+            NTE_BAD_KEY = 0x80090003,
+            NTE_BAD_FLAGS = 0x80090009,
+            NTE_BAD_DATA = 0x80090005,
+            NTE_BAD_HASH = 0x80090003,
+        }
+
+        /// <summary>
+        /// dwProvType is fourth parameter in the CryptAcquireContext and it uses following
+        /// </summary>
+        internal enum ProviderType : int
+        {
+            PROV_RSA_FULL = 1,
+            PROV_DSS_DH = 13,
+            PROV_RSA_AES = 24
+        }
+
+        /// <summary>
+        /// Helper for RsaCryptoServiceProvider.ImportParameters()
+        /// </summary>
+        internal static byte[] ToKeyBlob(this RSAParameters rsaParameters, int algId)
+        {
+            // The original FCall this helper emulates supports other algId's - however, the only algid we need to support is CALG_RSA_KEYX. We will not
+            // port the codepaths dealing with other algid's.
+            if (algId != CapiHelper.CALG_RSA_KEYX)
+                throw new PlatformNotSupportedException();
```

Comment:
Yes please that would be valuable to me and customers. it would be great to not have any parameterless PNSE thrown. (Anywhere in crypto but at least whatever you're touching)